stream-chat-android-client/src/test/java/io/getstream/chat/android/client/socket/experimental/ws/StreamWebSocketTest.kt

```diff
@@ -0,0 +1,167 @@
+    private val streamWebSocket = StreamWebSocket(parser) {
+        webSocketListener = it
+        webSocket
+    }
+
+    @Test
+    fun `When a chatEvent is sent, websocket should send a parsed event`() {
+        val chatEvent = mock<ChatEvent>()
+        val textEvent = randomString()
+        val expectedResult = randomBoolean()
+        whenever(parser.toJson(chatEvent)) doReturn textEvent
+        whenever(webSocket.send(textEvent)) doReturn expectedResult
+
+        val result = streamWebSocket.send(chatEvent)
+
+        verify(webSocket).send(textEvent)
+        result `should be equal to` expectedResult
+    }
+
+    @Test
+    fun `When close is called, the internal socket should be called`() {
+        val expectedResult = randomBoolean()
+        whenever(webSocket.close(closeSocketCode, closeSocketReason)) doReturn expectedResult
+
+        val result = streamWebSocket.close()
+
+        verify(webSocket).close(closeSocketCode, closeSocketReason)
+        result `should be equal to` expectedResult
+    }
+
+    @Test
+    fun `When messages are received by websocket, they should be added to eventFlow`() = runTest {
+        val eventsMap = List(positiveRandomInt(50)) { mock<ChatEvent>() }.associateBy { randomString() }
+        eventsMap.forEach { textEvent, chatEvent ->
+            whenever(parser.fromJsonOrError(textEvent, ChatEvent::class.java)) doReturn Result.success(chatEvent)
+        }
+        val listener = streamWebSocket.listen()
+        val listResult = mutableListOf<StreamWebSocketEvent>()
+        val job = launch { listener.collect { listResult.add(it) } }
+        runCurrent()
+
+        eventsMap.keys.forEach { textEvent -> webSocketListener.onMessage(webSocket, textEvent) }
+        runCurrent()
+
+        listResult `should be equal to` eventsMap.values.map { StreamWebSocketEvent.Message(it) }
+        job.cancel()
+    }
+
+    @Test
+    fun `When an known error is received by websocket, it should be added to eventFlow`() = runTest {
+        val textEvent = randomString()
+        val code = randomInt()
+        val message = randomString()
+        val statusCode = randomInt()
+        val errorResult = Result.error<ChatEvent>(mock<Throwable>())
+        whenever(parser.fromJsonOrError(textEvent, ChatEvent::class.java)) doReturn errorResult
+        whenever(parser.fromJsonOrError(textEvent, SocketErrorMessage::class.java))
+            .doReturn(Result.success(SocketErrorMessage(ErrorResponse(code, message, statusCode))))
+        // val listener = streamWebSocket.listen()
+        // val listResult = mutableListOf<StreamWebSocketEvent>()
+        // val job = launch { listener.collect { listResult.add(it) } }
+        // runCurrent()
+        //
+        // webSocketListener.onMessage(webSocket, textEvent)
+        // runCurrent()
+        //
+        // listResult `should be equal to` listOf(StreamWebSocketEvent.Error(ChatNetworkError.create(code, message, statusCode)))
+        // job.cancel()
+    }
```

Comment:
I wonder if comments are left intentionally

Reply:
Ups, nop, it was because some error I had with the mocking library.
Let me uncomment it

Reply:
Fixed!!!